### ISSUE TITLE
IC tests have too tight tolerances on CUDA

### ISSUE THREAD

**Issue by Olllom** [noegroup/bgflow#17]
Current main branch (commit 94fb9c1a6741878e1440955a01105505d2431028):
CUDA 11.2
```
FAILED tests/nn/flow/crd_transform/test_ic.py::test_angle_deriv[cuda:0-dtype0] - AssertionError: assert False
FAILED tests/nn/flow/crd_transform/test_ic.py::test_torsion_deriv[cuda:0-dtype0] - AssertionError: assert False
FAILED tests/nn/flow/crd_transform/test_ic.py::test_ic2xyz_deriv[cuda:0-dtype0] - AssertionError: assert False
FAILED tests/nn/flow/crd_transform/test_ic.py::test_global_ic_transform[cuda:0-dtype0] - AssertionError: normalize_angles=True;bonds != bonds_recon
FAILED tests/nn/flow/crd_transform/test_ic.py::test_global_ic_inversion[dtype0-cuda:0] - AssertionError: assert False
FAILED tests/nn/flow/crd_transform/test_ic.py::test_mixed_ic_inversion[dtype0-cuda:0] - AssertionError: assert False
```

**Comment by jonas-koehler** [comment]
Is this fixed by simply setting tolerances higher? Or is this a systematic error indicating some fundamental problem? Can you report concrete numbers where those assertions fail?

**Comment by Olllom** [comment]
It does not look like a systematic problem. If I loosen tolerances by one order of magnitude, the problem goes away. Can you reproduce it on CUDA?

**Comment by Olllom** [comment]
fixed in #27
